clients/da-vinci-client/src/main/java/com/linkedin/davinci/DaVinciBackend.java

```diff
@@ -160,15 +160,30 @@ public DaVinciBackend(
       IsolatedIngestionUtils.destroyLingeringIsolatedIngestionProcess(configLoader);
       /**
        * The constructor of {@link #storageService} will take care of unused store/store version cleanup.
+       *
+       * When Ingestion Isolation is enabled, we don't want to restore data partitions here:
+       * 1. It is a waste of effort since all the opened partitioned will be closed right after.
+       * 2. When DaVinci memory limiter is enabled, currently, SSTFileManager doesn't clean up the entries belonging
+       *    to the closed database, which means when the Isolated process hands the database back to the main process,
+       *    some removed SST files (some SST files can be removed in Isolated Process because of log compaction) will
+       *    remain in SSTFileManager tracked file list.
+       * 3. We still want to open metadata partition, otherwise {@link StorageService} won't scan the local db folder.
+       *    Also opening metadata partition in main process won't cause much side effect from DaVinci memory limiter's
+       *    POV, since metadata partition won't be handed back to main process in the future.
+       * 4. When Ingestion Isolation is enabled with suppressing live update feature, main process needs to open all the
+       *    data partitions since Isolated Process won't re-ingest the existing partitions.
        */
+      boolean whetherToRestoreDataPartitions = !isIsolatedIngestion()
+          || configLoader.getVeniceServerConfig().freezeIngestionIfReadyToServeOrLocalDataExists();
+      LOGGER.info("DaVinci {} restore data partitions.", whetherToRestoreDataPartitions ? "will" : "won't");
       storageService = new StorageService(
           configLoader,
           aggVersionedStorageEngineStats,
           rocksDBMemoryStats,
           storeVersionStateSerializer,
           partitionStateSerializer,
           storeRepository,
-          true,
+          whetherToRestoreDataPartitions,
```

Comment:
I guess I get what you mean here, also there is another logic at the end of the bootstrap trying to close metadata partition or metadata + data partition.
btw if we don't restore data partition, how can we drop storage engine cleanly? Is it done by forcibly drop whole folder?

Reply:
If there is no dvc memory limiter, the data partition will be either opened by the II process or by the main process, and the forcibly drop the database is mainly for the partitions, which are not opened by any process.